sandbox/modules/identity/src/main/java/org/opensearch/identity/authz/PermissionStorage.java

```diff
@@ -26,33 +29,57 @@
  */
 public class PermissionStorage {
 
-    public HashMap<Principal, List<Permission>> permissionStore = new HashMap<>();
+    public static HashMap<Principal, List<OpenSearchPermission>> permissionStore = new HashMap<>();
 
     /**
      * This function adds a set of permissions to the permission store. The principal is a unique identifier for some subject.
      * The List is a list of all permissions that are being granted.
      */
-    public void put(Principal principal, List<Permission> permissions) {
+    public static void put(Principal principal, List<OpenSearchPermission> permissions) {
 
-        permissionStore.put(principal, permissions);
+        for (OpenSearchPermission permission : permissions) {
+            permissionStore.computeIfAbsent(principal, permissionsList -> new ArrayList<>()).add(permission);
```

Comment:
It is the accepted value of SO so it seemed like a good option to do this in one line: https://stackoverflow.com/questions/12134687/how-to-add-element-into-arraylist-in-hashmap

Reply:
Can this be changed to `permissionStore.computeIfAbsent(principal,() -> new ArrayList<>()).add(permission);` or `permissionStore.computeIfAbsent(principal,(p) -> new ArrayList<>()).add(permission);`? The param is not getting used so it can be shortened/empty.

Reply:
Will do.